Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -117,6 +121,57 @@ public int ComponentIndex
             }
         }
 
+#if UNITY_EDITOR
+        void OnValidate()
+        {
+            Debug.Log($"NetworkBehaviour:OnValidate for {gameObject.name}");
+
+            List<NetworkIdentity> networkIdentities = transform.root.GetComponentsInChildren<NetworkIdentity>().ToList();
```

Comment:
not sure how I feel about that.
couldn't NetworkIdentity.cs OnValidate just check if any children have NetworkComponents?
that seems way more straight forward

Reply:
Reset fires first, and in NI.Reset I have added code to prevent the NI being added by RequireComponent if another NI exists.  If I do that in OnValidate (fires after it's added), it's too late and I'll get a deadlock of "can't remove" because of the dependency requirement of the other component.